### PR TITLE
Repair broken links in Markdown

### PR DESCRIPTION
**Pull request by cjduncana**
This PR is ready to merge.
This resolves https://github.com/roc-lang/roc/issues/3832

### CODE REVIEW

File: mlc_config.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            "pattern": "http://localhost"
+        },
+        {
+            "pattern": "http://moleorbataque.io"
+        },
+        {
+            "pattern": "http://occideratparem.io"
+        },
+        {
+            "pattern": "http://pedesque-huius.org"
```
Comment (folkertdev) [inline mlc_config.json:37]: where is this list from? `localhost` makes sense, but the rest?

File: .github/workflows/markdown_link_check.yml [hunk 2]
```diff
@@ -15,5 +15,4 @@ jobs:
       with:
         use-quiet-mode: 'yes'
         use-verbose-mode: 'yes'
-        check-modified-files-only: 'yes'
```
Comment (Anton-4) [inline .github/workflows/markdown_link_check.yml:18]: We probably want to keep this 'yes' right?

File: mlc_config.json (hunk 1 above)
Comment (Anton-4) [reply mlc_config.json:37]: Looks like these are generated by the lorem ipsum generator and ended up here https://github.com/roc-lang/roc/blob/main/examples/static-site-gen/input/apple.md . We should probably replace all those with a link to the roc website or something. Allowing a bunch of random links in the repo does not set a good precedent.

**Review by Anton-4 — CHANGES_REQUESTED** [review]

File: mlc_config.json (hunk 1 above)
Comment (Anton-4) [reply mlc_config.json:37]: I'll take care of that.

File: .github/workflows/markdown_link_check.yml (hunk 2 above)
Comment (cjduncana) [reply .github/workflows/markdown_link_check.yml:18]: We can keep it, but we will miss that check if a link dies for another reason unrelated to a code change.

Comment (Anton-4) [reply .github/workflows/markdown_link_check.yml:18]: Right, I see. Deleting it is best then.

**Review by Anton-4 — APPROVED** [review]

**Pull request merged**